### PR TITLE
contributing guide added

### LINKED ISSUE #94: Create a guide for contributing
Add a guide to contributing file compiling a list of points people need to keep in mind while adding bugs or patches to the repository.
This can be added as a `CONTRIBUTING.md` file in the `.github/` folder

Let us discuss some points to keep in mind here after which someone can be assigned the task of creating a guide.

### PR DESCRIPTION
**Pull request by nityanandagohain**
Fixes #94 

### Checklist
- [x] My branch is up-to-date with upstream/develop branch.
- [x] Everything works and tested for Python 3.5.2 and above.
- [x] I have created/claimed the issue that this PR resolves.

### Description

I have added CONTRIBUTING.md file

### CODE REVIEW

**Comment by chrizandr** [comment]
Few more points that could be added to this are(I leave it to you to decide which section they best fit in):

- Always ensure that you are working on an issue that has been assigned to you. Assigning issues ensures that two people are not separately working on the same thing.
- Please get feedback on issues before you start working on them. Do not work on an issue that has not been confirmed to be an issue by org members.
- In case you are not sure if the issue is an actual issue or not, please add the `question` label to it.
- Check if an issue has already been resolved before creating an issue. Please also ensure that you go through the docs and the wiki before you ask a question.
- If more than one person is willing to work on an issue, the work can be divided between them. This will also help to get an issue resolved much faster.
- Please squash your commits before making a PR. This ensures that all bug fixes to a single issue are in one commit. You can read more about squashing commits [https://gist.github.com/patik/b8a9dc5cd356f9f6f980](here).
- Ensure that you run all tests on your branch before making a pull request. Please make sure all tests pass before you create a Pull Request. In case tests are failing, you can ask others for help or create an issue about it.

@xadahiya @Mec-iS Anything you would like to add?

**Comment by nityanandagohain** [comment]
Thanks @chrizandr . I will wait for @xadahiya @Mec-iS suggestions.

**Comment by xadahiya** [comment]
Looks good to me 👍

**Comment by Mec-iS** [comment]
> Always ensure that you are working on an issue that has been assigned to you. Assigning issues ensures that two people are not separately working on the same thing.

This is wrong. Everybody should be able to fork and work on whatever without any assignment. The time at which different code from different people is adjusted is at PR time. We can have more PRs for the same issue and let people make adjustments by confronting the PRs. Any reference about asking for assignment has to be deleted, I didn't write it in the guidelines and I don't know who came up with this.

Also please avoid to write too many rules. We want to give *guidelines* not an enterprise rulebook.

**Comment by nityanandagohain** [comment]
@Mec-iS I have filtered it and made it shorter . Please review

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+### Contribution Best Practices
+
+**Feature Requests and Bug Reports**
```
Comment (Mec-iS) [inline .github/CONTRIBUTING.md]: This section is not useful

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+* Some newcomers are not accustomed to issue trackers and might try to inform developers on the mailing list, 
+chat or even write private emails about bugs and features, but the right place to do this is: The issue tracker of a project.
+
+* Any bug or feature, please open an issue in the issue tracker right away and indicate if you want to work on it yourself.
+
+* When you file a bug report on the issue tracker, make sure you add steps to reproduce it. Especially if that bug is some weird/rare one.
+
+* In case you are not sure if the issue is an actual issue or not, please add the question label to it.
+
+**Working on issues**
```
Comment (Mec-iS) [inline .github/CONTRIBUTING.md]: This section is not useful. Most of the things are already defined in Github Workflow.

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (Mec-iS) [reply .github/CONTRIBUTING.md]: I don't see any of the guidelines specified [here](https://github.com/HTTP-APIs/hydrus/wiki/Starting-Material#general-guidelines-to-contributing) mentioned.

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (Mec-iS) [reply .github/CONTRIBUTING.md]: Text should be kept short and practical. No bureaucratic idioms.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+* Read all the texts related to [contributing for an OS community](https://github.com/HTTP-APIs/hydrus/tree/master/.github).
+
+* Read this [how-to about writing a PR](https://github.com/blog/1943-how-to-write-the-perfect-pull-request) and this [other how-to about writing a issue](https://wiredcraft.com/blog/how-we-write-our-github-issues/)
+
+* If you find a problem, first ask for [help in the chat](https://gitter.im/HTTP-APIs/Lobby), then consider opening a issue.
+    
+* Before opening a PR be sure that all the tests pass successfully. If any is failing for non-related reasons, annotate the test failure in the PR comment.
+
+* Any change should be PRed first in ```develop```, ```master``` can only receive merge from develop.
```
Comment (Mec-iS) [inline .github/CONTRIBUTING.md]: Triple ticks ``` are meant for multiline strings. For single line strings one tick is the md to use:
* `single`
*
```
multiline
multiline
...
```

Comment (Mec-iS) [reply .github/CONTRIBUTING.md]: The rest is ok.

**Pull request merged**